atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/GetAsyncDelegate.java

```diff
@@ -41,11 +51,10 @@ public Transaction delegate() {
     @Override
     public Map<Cell, byte[]> get(TableReference tableRef, Set<Cell> cells) {
         try {
-            return super.getAsync(tableRef, cells).get();
-        } catch (InterruptedException e) {
-            throw Throwables.rewrapAndThrowUncheckedException(e);
-        } catch (ExecutionException e) {
-            throw Throwables.rewrapAndThrowUncheckedException(e.getCause());
+            enterFunction.run();
+            return AtlasFutures.getUnchecked(delegate().getAsync(tableRef, cells));
+        } finally {
+            exitFunction.run();
```

Comment:
This is possibly correct, but is a bit of a non-standard pattern: if `enterFunction.run()` throws, do we still want `exitFunction` to run? This depends on the semantics you want. Since it's tests it probably doesn't matter too much.

If we adopt Felix's suggestion (which I agree with), then this is a non-issue.

Reply:
In general we don't expect this `enterFunction.run()` to throw, now it should be clearer with the refactoring also curious about what you think about this specific part.